valhalla/meili/config.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <boost/property_tree/ptree.hpp>
+
+namespace valhalla {
+namespace meili {
+
+struct Config {
+  // create config with default parameters
+  Config() = default;
+  // read parameters from boost::ptree (use default values for missing params)
+  explicit Config(const boost::property_tree::ptree& params);
+  // override default config parameters
+  void Read(const boost::property_tree::ptree& params);
+
+  struct CandidateSearch {
+    // default search radius (meters) for measurements; it's used to determine maximum search radius
+    float search_radius = 50.f;
+    // maximum allowed difference (meters) between GPS point and corresponding route point
+    float max_search_radius = 200.f;
+
+    size_t cache_size = 100240;
+    size_t grid_size = 500;
+
+    void Read(const boost::property_tree::ptree& params);
+  };
+
+  struct Transition {
+    // beta parameter of exponential distribution
+    float beta = 3.f;
+    // maximum allowed difference (meters) between route distance and great circle distance
+    float breakage_distance = 2000.f;
+    // maximum allowed ratio of route distance to great circle distance
+    float max_route_distance_factor = 5.f;
+    // maximum allowed ratio of route time distance to the corresponding measurements time distance
+    float max_route_time_factor = 5.f;
+    // it's used to determine 'turn cost' depending on turn degree
+    float turn_penalty_factor = 200.f;
+
+    void Read(const boost::property_tree::ptree& params);
+  };
+
+  struct Emission {
```

Comment:
This struct specifically characterizes a likelihood function, so I'd call it `EmissionLikelihood`. Same for `TransitionLikelihood`.

Reply:
It should probably be `TransitionCost`, etc. since that's existing terminology used in meili.